File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> arr) {
    int answer = 0;
    vector<int> num(4);
    bool p;
    int i = 1;
    
    sort(arr.begin(), arr.end());
    
    while (1) {
        p = true;
        answer = arr.back() * i;
        
        for (int i = 0; i < arr.size(); i++) {
            if (answer % arr[i] != 0) {
                p = false;
                break;
            }
        }
        
        if (p) break;
        i++;
    }
    
    return answer;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.kt
class Solution {
    fun solution(arr: IntArray): Int {
        var answer = 0
        var i = 1
        var p = true
        
        while(true) {
            p = true
            answer = arr.last() * i
            
            for (item in arr) {
                if (answer % item != 0) {
                    p = false
                    break;
                }
            }
            
            if (p) break;
            
            i += 1
        }
        
        return answer
    }
}